packages/x-data-grid/src/components/quickFilter/QuickFilter.tsx

```diff
@@ -81,7 +81,9 @@ function QuickFilter(props: QuickFilterProps) {
   const setQuickFilterValueDebounced = React.useMemo(
     () =>
       debounce((newValue: string) => {
-        apiRef.current.setQuickFilterValues(parser(newValue));
+        const newQuickFilterValues = parser(newValue);
+        prevQuickFilterValuesRef.current = newQuickFilterValues;
```

Comment:
Restores the logic from the original GridToolbarQuickFilter, as seen here in a version of the code before I merged the toolbar update: https://github.com/mui/mui-x/blob/7c80083adc800324accee8288cb6c4a58c033f55/packages/x-data-grid/src/components/toolbar/GridToolbarQuickFilter.tsx#L119